### PR TITLE
fix(kit): correct detect iso code for Kazakhstan & correct limit in the field for entering a phone number

### PR DESCRIPTION
**Pull request by mdlufy**
## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behavior?
1. the limit of 12 characters in the field for entering a phone number
2. when inserting a number with the code +7, the country code changes to Kazakh

Closes #4217

### CODE REVIEW

File: projects/kit/components/input-phone-international/input-phone-international.component.ts [hunk 1]
```diff
@@ -151,15 +154,20 @@ export class TuiInputPhoneInternationalComponent
...
+        );
 
         if (!countryIsoCode) {
-            this.updateValue(
-                `${this.inputPhoneCountryCode}${value}`
-                    .replace(TUI_MASK_SYMBOLS_REGEXP, '')
-                    .slice(0, this.getMaxAllowedLength(this.countryIsoCode)),
-            );
+            // @ts-ignore
+            this.value = `${this.inputPhoneCountryCode}${value}`
```
Comment (nsbarsukov) [inline projects/kit/components/input-phone-international/input-phone-international.component.ts:165]: `AbstractTuiControl` doesn't have this setter in `v2.x`.
In `2.x` you should use `updateValue(...)`.

___
Setter `value` of `AbstractTuiControl` was released only in 3.23.0:
* https://github.com/taiga-family/taiga-ui/pull/4048

Comment (mdlufy) [reply projects/kit/components/input-phone-international/input-phone-international.component.ts]: thanks, fixed it

File: projects/kit/pipes/iso-to-country-code/iso-to-country-code.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import {Inject, Pipe, PipeTransform} from '@angular/core';
+import {TuiCountryIsoCode} from '@taiga-ui/i18n';
+import {TUI_COUNTRIES_MASKS} from '@taiga-ui/kit/tokens';
+import {tuiIsoToCountryCode} from '@taiga-ui/kit/utils';
+
+@Pipe({
```
Comment (nsbarsukov) [inline projects/kit/pipes/iso-to-country-code/iso-to-country-code.pipe.ts:6]:
```suggestion
// @dynamic
@Pipe({
```

File: projects/kit/pipes/to-country-code/to-country-code.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import {CHAR_PLUS} from '@taiga-ui/cdk';
+import {TuiCountryIsoCode} from '@taiga-ui/i18n';
+import {TUI_COUNTRIES_MASKS} from '@taiga-ui/kit/tokens';
+import {
+    tuiGetMaxAllowedPhoneLength,
+    tuiIsoToCountryCode,
+    tuiNotKzRegion,
+} from '@taiga-ui/kit/utils';
+
+@Pipe({
```
Comment (nsbarsukov) [inline projects/kit/pipes/to-country-code/to-country-code.pipe.ts:11]:
```suggestion
// @dynamic
@Pipe({
```

Comment (mdlufy) [reply projects/kit/pipes/to-country-code/to-country-code.pipe.ts:12]: thanks!

File: projects/kit/pipes/iso-to-country-code/iso-to-country-code.pipe.ts (hunk 2 above)
Comment (mdlufy) [reply projects/kit/pipes/iso-to-country-code/iso-to-country-code.pipe.ts:7]: thanks!

**Review by nsbarsukov — APPROVED** [review]

**Review by splincode — APPROVED** [review]

**Pull request merged**